src/node/node.c

```diff
@@ -338,16 +341,22 @@ static void set_value(struct s_plugin_conf *conf, const char *key,
 	size_t i;
 	size_t key_len = strlen(key);
 
+	if (key_len + strlen(value) + 1 >= MAX_ENV_BUF_SZ) {
+		fprintf(stderr, "env var key+value too long: %.30s...\n",
+			key);
+		abort();
+	}
+
 	struct s_env *dst_env = NULL;
 	/* Search for the corresponding env */
-	for (i = 0; i < conf->size; i++) {
+	for (i = 0; i < conf->used; i++) {
 		struct s_env *env = conf->env + i;
 
 		if (key_len != env->key_len)
 			continue;
 
 		/* this cmp works since keys have the same length */
-		if (strncmp(key, env->buffer, env->key_len) != 0)
+		if (memcmp(key, env->buffer, env->key_len) != 0)
```

Comment:
Don't we need to also copy the `\0` ? 
I'm not sure `realloc` initialize to zero.

Reply:
This is just a compare - equivalent to `strncmp()` but arguably a little simpler. We know that the new key is key_len and we know that the key in `env->buffer` is also key_len, and after that is a `=` instead of a `\0`. So the `strncmp()` was to stop the compare just before the `=` in the buffer. `strncmp()` would also stop at `\0` but we already know that's past `key_len` anyway, so we can use `memcmp()`. Or I can put it back :)

Reply:
Sorry, I needed a ☕ this morning. 
I read too quickly and mistook it for a `memcpy()` ... 😞 

All good!